pkg/querier/querier.go

```diff
@@ -652,7 +653,7 @@ func validateQueryTimeRangeLimits(ctx context.Context, userID string, limits tim
 		return time.Time{}, time.Time{}, httpgrpc.Errorf(http.StatusBadRequest, util_validation.ErrQueryTooLong, (through).Sub(from), maxQueryLength)
 	}
 	if through.Before(from) {
-		return time.Time{}, time.Time{}, httpgrpc.Errorf(http.StatusBadRequest, "invalid query, through < from (%s < %s)", through, from)
+		return time.Time{}, time.Time{}, httpgrpc.Errorf(http.StatusBadRequest, "this data is no longer available, it is past now - max_query_lookback (%s)", maxQueryLookback)
```

Comment:
WDYT of somehow merging both messages? this way users filtering logs based on the older message won't experience this as a breaking change

Reply:
I'd rather not because the old message is just wrong/deceptive. It changes the from parameter and prints the new value.

Reply:
ohhh you right. I think this looks good so I'll resolve the thread, but I'll let to others to approve just because, even though the previous message is misleading, it is still a change in an HTTP request return, which I'm not sure how impactful it can be